File: hardlevel/STRING-PERMUTATIONS/String-Premutations.cs
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            Showpermutations(line);
        }
    }
    
    public class myOrderClass : IComparer  {
        int IComparer.Compare( Object x, Object y )  
        {
            string X = (string)x;
            string Y = (string)y;
            return string.CompareOrdinal(X, Y);
        }
    
    }
    
    public static void Showpermutations(string line) {
        string result = "";
        int size = line.Length;
        if (size > 0) {
            ArrayList myList = new ArrayList();
            getPermutations("", line, myList);
            myList.Sort(new myOrderClass());
            foreach(string word in myList) result += word + ",";
            result = result.Substring(0, result.Length - 1);
        }
        Console.WriteLine(result);
    }

    public static void getPermutations(string beginningString,string endingString, 
                                        IList myList) {
        if (endingString.Length <= 1) {
            myList.Add(beginningString + endingString);
        } else {
                for (int i = 0; i < endingString.Length; i++) {
                    string newString = endingString.Substring(0, i) + 
                                        endingString.Substring(i + 1);
                getPermutations(beginningString + endingString[i], 
                                newString, myList);
            }
        }
    }
}

File: hardlevel/STRING-PERMUTATIONS/String-Premutations.java
import java.io.*;
import java.util.ArrayList;
import java.util.Collections;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            // Process line of input Here
            Showpermutations(line);
        }
    }
    
    public static void Showpermutations(String line) {
        String result = "";
        int size = line.length();
        if (size > 0) {
            ArrayList<String> myList = new ArrayList();
            getPermutations("", line, myList);
            Collections.sort(myList);
            for (String word:myList) result = result.concat(word+",");
            result = result.substring(0, result.length() - 1);
        }
        System.out.println(result);
    }

    public static void getPermutations(String beginningString, String endingString, ArrayList<String> myList) {
        if (endingString.length() <= 1) {
            myList.add(beginningString + endingString);
        } else {
            for (int i = 0; i < endingString.length(); i++) {
                String newString = endingString.substring(0, i) + endingString.substring(i + 1);
                getPermutations(beginningString + endingString.charAt(i), newString, myList);
            }
        }
    }
}
